test/integration/http/websockets-test.js

```diff
@@ -1,68 +1,100 @@
 let { join } = require('path')
 let test = require('tape')
-let websocket = require('ws')
+let Websocket = require('ws')
 let sut = join(process.cwd(), 'src')
 let sandbox = require(sut)
-let { data, shutdown } = require('./_utils')
+let { makeSideChannel } = require('./_utils')
 
 let mock = join(process.cwd(), 'test', 'mock')
 let url = `ws://localhost:${process.env.PORT || 3333}`
-let wsErr = err => { throw err }
-let ws // Reused below
-
-// Lots of test fns calling other test fns in this test
-function setup (t) {
-  if (ws) throw Error('Should not have found WebSocket, test is not clean')
-  ws = new websocket(url)
-  ws.on('error', wsErr)
-  t.ok(ws, 'WebSocket set up')
+let _ws
+let _events
+
+let connectWebSocket = async () => {
+  if (_ws) throw Error('Only one websocket can be connected at a time, test is not clean')
+  _ws = new Websocket(url)
+  _ws.on('error', err => { throw err })
+  _ws.on('close', () => {
+    _ws = undefined
+    console.log('WebSocket disconnected')
+  })
+  await new Promise((resolve) => _ws.on('open', resolve))
```

Comment:
Any feels on `promisify`ing all these?

Reply:
You know, now that you mention it, no idea! Hah. Maybe `promisify(ws.on.bind({}, 'open'))`? idk! Feel free to ignore me. :)